specs/spec.go

```diff
@@ -73,9 +74,21 @@ func (s *Spec) UnmarshalJSON(data []byte) error {
 	default:
 		return fmt.Errorf("unknown kind %s", s.Kind)
 	}
-	b, err := json.Marshal(t.Spec)
-	if err != nil {
-		return err
+	var b []byte
+	var err error
+	// nolint:all
+	if t.Spec != nil {
+		b, err = json.Marshal(t.Spec)
+		if err != nil {
+			return err
+		}
+	} else if t.Plugin != nil {
+		b, err = json.Marshal(t.Plugin)
+		if err != nil {
+			return err
+		}
+	} else {
+		return fmt.Errorf("plugin is required and shouldn't be empty")
```

Comment:
This check wasn't here before, new functionality?

Reply:
yeah I think it was a bug prob but can go either way